apps/backend/src/controllers/api/userRouter.ts

```diff
@@ -14,35 +16,77 @@ userRouter.post(
   "/",
   async (req: Request, res: Response, next: NextFunction) => {
     try {
-      const { email, username, password } = req.body;
-      if (!email.length) {
-        res.status(400).send({ error: "Your email cannot be blank" });
+      const { email, username, password, oauth, role } = req.body;
+      const cleanedEmail = email?.trim();
+      const cleanedUsername = username?.trim();
+      const cleanedPassword = password?.trim(); 
+      const cleanedRole = role?.trim();
+
+      if (!cleanedEmail?.length) {
+        res.status(400).send({ error: "Your email cannot be blank." });
+        return;
+      }
+
+      if (!cleanedUsername?.length) {
+        res.status(400).send({ error: "Your username cannot be blank." });
+        return;
+      }
+
+      if (oauth === undefined && !cleanedPassword?.length) {
+        res.status(400).send({ error: "Your password cannot be blank." });
         return;
       }
 
-      if (!password.length) {
-        res.status(400).send({ error: "Your password cannot be blank" });
+      if (!cleanedRole?.length) {
+        console.log(cleanedRole);

```

Comment:
```suggestion
        logger.info(`A blank role was given! Cleaned role: ${cleanedRole}`);
```
Can clean up the console logs that are lying around so that the backend logs don't get too clogged up. Or if you want to keep them then format it into a cleaner message like above and using the logger defined in the backend service.